### PR TITLE
Fix bugs in the merge parameter strategy for added and dropped parame…

### PR DESCRIPTION
**Pull request by pdrakeweb**
…ters.

### CODE REVIEW

**Comment by gergelyborsothy** [comment]
Cool, this behaviour was actually not expected (by me at least) for some reason, so totally makes sense. 👍

**Comment by glennpratt** [comment]
👍

File: lib/moonshot/stack.rb [hunk 1]
```diff
@@ -138,7 +138,7 @@ def resources_of_type(type)
     # update.
     def overrides
       if File.exist?(parameters_file)
-        YAML.load_file(parameters_file)
+        YAML.load_file(parameters_file) || {}
```
Comment (askreet) [inline lib/moonshot/stack.rb]: When does this return falsey? If the yaml file is "---\nfalse"?

File: docs/user-guide/stack_parameter_strategies.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ Example:
 
 ```ruby
 class CustomStrategy
-  def parameters(current, overrides)
+  def parameters(parameters, stack_parameters, template)
     current.map do |k, v|
```
Comment (codwazny) [inline docs/user-guide/stack_parameter_strategies.md]: Current isn't defined anymore.

File: lib/moonshot/default_strategy.rb [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+    # Default strategy: use parameter defined in the parameter file
     class DefaultStrategy
-      def parameters(_, overrides)
-        overrides.map do |key, _|
+      def parameters(params, _, _)
+        params.map do |key, _|
           {
             parameter_key: key,
-            parameter_value: overrides[key],
+            parameter_value: params[key],
```
Comment (codwazny) [inline lib/moonshot/default_strategy.rb]: Any reason for doing this over having the map block yield value and putting value here? Pretty sure params[key] is just going to end up doing another hash look up for no reason.

File: lib/moonshot/stack.rb (hunk 1 above)
Comment (codwazny) [inline lib/moonshot/stack.rb]: Also returns false if the yaml file is empty but the file exists.

File: lib/moonshot/default_strategy.rb (hunk 3 above)
Comment (glennpratt) [inline lib/moonshot/default_strategy.rb]: I agree something like that would be easier to read, though hash lookups are super fast, so probably not a big performance concern.

Could also do "safe assignment in condition":

```ruby
if (value = params[key])

Comment (codwazny) [inline lib/moonshot/default_strategy.rb]: I think this comment is referring to the line 15 of merge_strategy.rb but is still relevant :)

Comment (glennpratt) [inline lib/moonshot/default_strategy.rb]: Uhh, wow, yes it is.

File: lib/moonshot/stack.rb (hunk 1 above)
Comment (pdrakeweb) [inline lib/moonshot/stack.rb]: If the YAML file contains valid YAML for nil ("---").

File: lib/moonshot/default_strategy.rb (hunk 3 above)
Comment (pdrakeweb) [inline lib/moonshot/default_strategy.rb]: Only because its how this is already done and I was just renaming the variable to make it more understandable.

**Pull request merged**